### PR TITLE
fix UnboundLocalError in tune.run (#142)

### PR DESCRIPTION
**Pull request by eduardobull**
Fix UnboundLocalError exception in `tune.run` when `training_function` returns a value.

### CODE REVIEW

**Comment by eduardobull** [comment]
Resolves #142

**Review by sonichi — COMMENTED** [review]
Thanks. Could you add your test that covers this case in test/tune/test_tune.py?

File: test/tune/test_tune.py [hunk 1]
```diff
@@ -217,6 +217,22 @@ def simple_func(config):
...
 
+def test_run_training_function_return_value():
+    from flaml import tune
+
+    def evaluate_config(config):
+        metric = (round(config['x'])-85000)**2 - config['x']/config['y']
+        return {"metric": metric}
+
+    tune.run(
+        evaluate_config,
```
Comment (sonichi) [inline test/tune/test_tune.py:228]: The test failed. Either pass `metric='metric'` to tune.run or return a float in `evaluate_config`

**Review by sonichi — CHANGES_REQUESTED** [review]
Please sign

File: test/tune/test_tune.py (hunk 1 above)
Comment (eduardobull) [reply test/tune/test_tune.py]: I've modified the test function to pass a `training_function` returning a dict and another one returning a scalar.

**Review by sonichi — APPROVED** [review]

**Pull request merged**